src/trio/_tests/test_deprecate.py

```diff
@@ -254,9 +254,19 @@ def test_module_with_deprecations(recwarn_always: pytest.WarningsRecorder) -> No
     assert "value1 instead" in got.message.args[0]
 
     assert module_with_deprecations.dep2 == "value2"  # type: ignore[attr-defined]
-    got = recwarn_always.pop(TrioDeprecationWarning)
+    got = recwarn_always.pop(DeprecationWarning)
     assert isinstance(got.message, Warning)
     assert "instead-string instead" in got.message.args[0]
 
     with pytest.raises(AttributeError):
         module_with_deprecations.asdf  # type: ignore[attr-defined]  # noqa: B018  # "useless expression"
+
+
+def test_warning_class() -> None:
+    with pytest.warns(DeprecationWarning):
+        warn_deprecated("foo", "bar", issue=None, instead=None)
```

Comment:
For this (and quite a bit above), we can probably use specialized pytest helpers.

Reply:
this already uses `pytest.warns`, or what kind of helper do you mean?

Reply:
Specifically I was thinking of pytest.deprecated_call

Tbf this is more of a nitpick

Reply:
Haha, that is an incredibly nitty nitpick. Though we might care about whether we raise `DeprecationWarning` or `PendingDeprecationWarning` so I actually think the current implementation is better. We can do both though